bsp/Src/CAN.c

```diff
@@ -178,10 +178,11 @@ static const uint32_t can3_recv_entry_count = 0;
 #endif /* can3_recv_entries.h */
 #endif /* CAN3 */
 
-static inline void HAL_CAN_MspF4Init(CAN_HandleTypeDef* hcan){
+// CAN GPIO init, can be overridden by user
+__weak void HAL_CAN_MspGPIOInit(CAN_HandleTypeDef* hcan) {
   GPIO_InitTypeDef init = {0};
-  
   // CAN1
+  #ifdef CAN1
   if (hcan->Instance == CAN1) {
```

Comment:
if this whole if statement is just a comparison for hcan->Instance should/could this be a switch case?